### PR TITLE
Vite

### PR DESCRIPTION
**Pull request by ChrRubin**
Webpack slow. Now we Vite.

Uses [vite-plugin-web-extension](https://github.com/aklinker1/vite-plugin-web-extension), which simplifies a bunch of the manual entry work. The plugin also uses [web-ext](https://github.com/mozilla/web-ext) for dev mode, which will automatically launch the browser with Lofi Girl's stream, and handle extension reloads on code change.

However, there's a few things to consider before using this plugin for building LiveTL with vite:
- The plugin parses `manifest.json` to automatically set up entries and copy assets, and the build output tree will follow that of the manifest. This works fine for HyperChat, but on LTL everything HC related will be built to `build/submodules/chat/src/scripts` etc.
I guess if you REALLY want to you could move them out with a seperate plugin and edit the outputted manifest to reflect the new paths, but idk if that's more work than something like what we did on webpack.
- `vite-plugin-html-template` doesn't play nice with it, so each page needs their own HTML file.
- Svelte HMR doesn't work with the plugin (yet? HMR for HTML is on its roadmap at least).
- Extension pages will close immediately when the extension reloads on code change, which was already the behaviour for `webpack-ext-reloader`.
We can pass a `noReload` option so the extension doesn't automatically reload, but pages still need to be manually reloaded since there's no HMR, and content/background script changes require a manual extension reload.

A bunch of files were moved around, so we'd have to make changes on LTL side when updating the submodule:
- `assets` folder has been moved into `src`.
- Content and background scripts are moved into `scripts` so the build output makes more sense than a folder named `ts`.

### CODE REVIEW

**Comment by r2dev2** [comment]
Merge conflicts sadge. Can you fix them?

**Comment by ChrRubin** [comment]
> Merge conflicts sadge. Can you fix them?

Fixed

File: package.json [hunk 1]
```diff
@@ -3,56 +3,50 @@
   "version": "69.42.0",
   "private": true,
   "scripts": {
-    "build": "webpack --mode development",
-    "build:production": "webpack --mode production",
+    "build": "vite build",
```
Comment (r2dev2) [inline package.json:6]: This seems to error out when `build` folder does not exist. Can you make it create the folder before this with `mkdir -p build` or however you do that in node if we want to maintain cross-platform building?

**Review by r2dev2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ChrRubin) [reply package.json:6]: This is a bug on the current version of the plugin, it didn't error out on a previous version. I've opened an issue there and will see if they fix it.

Comment (r2dev2) [reply package.json:6]: I sent a pr to fix it at https://github.com/aklinker1/vite-plugin-web-extension/pull/11.

Comment (r2dev2) [reply package.json:6]: Pog, they merged. Can you upgrade now?

**Review by KentoNishi — APPROVED** [review]

**Pull request merged**